astroid/modutils.py

```diff
@@ -604,6 +604,12 @@ def is_relative(modname: str, from_file: str) -> bool:
     )
 
 
+@lru_cache(maxsize=1024)
+def isfile(path: str | os.PathLike) -> bool:
```

Comment:
Shouldn't it be 
```suggestion
def is_file(path: str | os.PathLike) -> bool:
```

To keep it in line with other utilities?

Reply:
Maybe even
```suggestion
def cached_is_file(path: str | os.PathLike) -> bool:
```